src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -254,3 +250,30 @@ function asyncLoad(url, onload, onerror) {
 function getArrayDataPointer<T>(array: System_Array<T>): number {
   return <number><any>array + 12; // First byte from here is length, then following bytes are entries
 }
+
+function attachInteropInvoker() {
+  const dotNetDispatcherInvokeMethodHandle = findMethod('Microsoft.JSInterop', 'Microsoft.JSInterop', 'DotNetDispatcher', 'Invoke');
+  const dotNetDispatcherBeginInvokeMethodHandle = findMethod('Microsoft.JSInterop', 'Microsoft.JSInterop', 'DotNetDispatcher', 'BeginInvoke');
+
+  DotNet.attachDispatcher({
+    beginInvokeDotNetFromJS: (callId, assemblyName, methodIdentifier, argsJson) => {
+      monoPlatform.callMethod(dotNetDispatcherBeginInvokeMethodHandle, null, [
+        callId ? monoPlatform.toDotNetString(callId.toString()) : null, // TODO: Implement a way to pass a number to a long without stringification
```

Comment:
Am still planning to do this "TODO" item.

Reply:
Changed my mind. The two approaches I considered both don't fit well.

 * Allocating the 8 bytes on the stack and passing directly as a raw value parameter: The existing code that prepares the params array on the stack assumes 4 bytes per param, and generalising it will come at a perf cost for every invocation, plus complicate the `Platform` interface considerably.
 * Passing as a boxed long: Mono doesn't currently expose any convenient APIs for allocating a heap object of an arbitrary type. And even if it did that's still a heap allocation, so the perf gain over passing as a string is not massive.

For now I'm keeping the "pass as string" behavior. It's not optimal for perf but it only affects this one code path. We can reconsider in the future if we decide it's consequential.